ark/type/__tests__/defaults.test.ts

```diff
@@ -364,4 +403,106 @@ contextualize(() => {
 			)
 		})
 	})
+
+	describe("factory functions", () => {
+		it("works in tuple", () => {
+			const t = type({ foo: ["string", "=", () => "bar"] })
+			attest(t.assert({ foo: "bar" })).snap({ foo: "bar" })
+		})
+		it("works in type tuple", () => {
+			const foo = type(["string", "=", () => "bar"])
+			const t = type({ foo })
+			attest(t.assert({ foo: "bar" })).snap({ foo: "bar" })
+		})
+		it("works in type args", () => {
+			const foo = type("string", "=", () => "bar")
+			const t = type({ foo })
+			attest(t.assert({ foo: "bar" })).snap({ foo: "bar" })
+		})
+	})
+
+	describe("works with objects", () => {
+		// it("default array in string", () => {
+		// 	const t = type({ bar: type("number[] = []") })
+		// 	attest(t.assert({}).bar).snap([])
+		// 	attest(t.assert({}).bar !== t.assert({}).bar)
+		// })
+		it("default array", () => {
+			const t = type({
+				foo: type("number[]").default(() => [1]),
+				bar: type("number[]")
+					.pipe(v => v.map(e => e.toString()))
+					.default(() => [1])
+			})
+			const v1 = t.assert({}),
+				v2 = t.assert({})
+			attest(v1).snap({ foo: [1], bar: ["1"] })
+			attest(v1.foo !== v2.foo)
+		})
+		it("default array is checked", () => {
+			attest(() => {
+				// @ts-expect-error
+				type({ bar: type("number[]").default(() => ["a"]) })
+				// THIS LOOKS WEIRD TBH
+			}).throws.snap(
+				"ParseError: Default value value at [0] must be a number (was a string)"
```

Comment:
this check's text is weird maybe needs fix

Reply:
Yeah maybe the prefix should just be "Default" instead of "Default value"